crates/server/src/server/request_handler.rs

```diff
@@ -70,16 +70,15 @@ impl Request {
         })
     }
 
-    /// Return just the header and request information from the Request Message
-    ///
-    /// Returns an error if there is not exactly one query
-    pub fn request_info(&self) -> Result<RequestInfo<'_>, ProtoError> {
-        Ok(RequestInfo {
+    /// Return just the header and request information from the Request Message.  Only the first
+    /// query in the request message is returned.
+    pub fn request_info(&self) -> RequestInfo<'_> {
+        RequestInfo {
             src: self.src,
             protocol: self.protocol,
             header: self.message.header(),
-            query: self.message.raw_queries().try_as_query()?,
-        })
+            query: &self.message.raw_queries().queries()[0],
```

Comment:
I think this change could benefit from more explanation of why it's assured that the `Vec` will never be empty. Seeing the raw indexing without more context makes me suspect the potential for a panic here.